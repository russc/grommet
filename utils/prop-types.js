"use strict";

exports.__esModule = true;
exports.backgroundPropType = exports.a11yTitlePropType = void 0;

var _reactDesc = require("react-desc");

var a11yTitlePropType = _reactDesc.PropTypes.string.description('Custom title to be used by screen readers.');

exports.a11yTitlePropType = a11yTitlePropType;

var backgroundPropType = _reactDesc.PropTypes.oneOfType([_reactDesc.PropTypes.string, _reactDesc.PropTypes.shape({
  color: _reactDesc.PropTypes.string,
  opacity: _reactDesc.PropTypes.oneOfType([_reactDesc.PropTypes.oneOf(['weak', 'medium', 'strong']), _reactDesc.PropTypes.bool])
})]).description('Background color');

exports.backgroundPropType = backgroundPropType;