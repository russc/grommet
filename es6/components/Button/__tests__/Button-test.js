import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { findAllByType } from '../../../utils';
import { Grommet, Button } from '../..';
test('Button renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    label: "Test",
    onClick: function onClick() {}
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button warns about invalid label render', function () {
  var warnSpy = jest.spyOn(console, 'warn');
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    label: "Test",
    onClick: function onClick() {}
  }, "invalid")));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(warnSpy).toHaveBeenCalledWith('Button should not have children if icon or label is provided');
  warnSpy.mockReset();
  warnSpy.mockRestore();
});
test('Button warns about invalid icon render', function () {
  var warnSpy = jest.spyOn(console, 'warn');
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    icon: React.createElement("svg", null),
    onClick: function onClick() {}
  }, "invalid")));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(warnSpy).toHaveBeenCalledWith('Button should not have children if icon or label is provided');
  warnSpy.mockReset();
  warnSpy.mockRestore();
});
test('Button primary renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    primary: true,
    label: "Test",
    onClick: function onClick() {}
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button color renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    color: "accent-1",
    label: "Test",
    onClick: function onClick() {}
  }), React.createElement(Button, {
    color: "accent-1",
    primary: true,
    label: "Test",
    onClick: function onClick() {}
  }), React.createElement(Button, {
    color: "#111111",
    primary: true,
    label: "Test",
    onClick: function onClick() {}
  }), React.createElement(Button, {
    color: "#123",
    primary: true,
    label: "Test",
    onClick: function onClick() {}
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button focus renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    focus: true,
    label: "Test",
    onClick: function onClick() {}
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button disabled renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    disabled: true
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button icon label renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    icon: React.createElement("svg", null),
    label: "Test",
    onClick: function onClick() {}
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button reverse icon label renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    reverse: true,
    icon: React.createElement("svg", null),
    label: "Test",
    onClick: function onClick() {}
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button href renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    href: "test"
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button hoverIndicator renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    onClick: function onClick() {},
    hoverIndicator: "background"
  }, "hoverIndicator")));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button hoverIndicator as object renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    onClick: function onClick() {},
    hoverIndicator: {
      background: true
    }
  }, "hoverIndicator")));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button hoverIndicator as object with color renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    onClick: function onClick() {},
    hoverIndicator: {
      background: 'brand'
    }
  }, "hoverIndicator")));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button hoverIndicator as object with colorIndex renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    onClick: function onClick() {},
    hoverIndicator: {
      background: 'accent-1'
    }
  }, "hoverIndicator")));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button hoverIndicator as object with invalid color renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    onClick: function onClick() {},
    hoverIndicator: {
      background: 'accent'
    }
  }, "hoverIndicator")));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    onClick: function onClick() {},
    hoverIndicator: {
      background: 'invalid'
    }
  }, "hoverIndicator")));
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button hoverIndicator as object with invalid colorIndex renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    onClick: function onClick() {},
    hoverIndicator: {
      background: 'accent-100'
    }
  }, "hoverIndicator")));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Button is clickable', function () {
  var onClick = jest.fn();
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(Button, {
    label: "Test",
    onClick: onClick
  })));
  var tree = component.toJSON();
  var button = findAllByType(tree, 'button');
  button[0].props.onClick();
  expect(onClick).toBeCalled();
});