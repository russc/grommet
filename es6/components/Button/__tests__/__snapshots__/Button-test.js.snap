// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button color renders 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #FD6FFF;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #FD6FFF;
}

.c3 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #FD6FFF;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background: #FD6FFF;
  color: #444444;
  border-radius: 18px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c3:hover {
  box-shadow: 0px 0px 0px 2px #FD6FFF;
}

.c4 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #111111;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background: #111111;
  color: #f8f8f8;
  border-radius: 18px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c4:hover {
  box-shadow: 0px 0px 0px 2px #111111;
}

.c5 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #123;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background: #123;
  color: #f8f8f8;
  border-radius: 18px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c5:hover {
  box-shadow: 0px 0px 0px 2px #123;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    <div
      className="c2"
    >
      Test
    </div>
  </button>
  <button
    className="c3"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    <div
      className="c2"
    >
      Test
    </div>
  </button>
  <button
    className="c4"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    <div
      className="c2"
    >
      Test
    </div>
  </button>
  <button
    className="c5"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    <div
      className="c2"
    >
      Test
    </div>
  </button>
</div>
`;

exports[`Button disabled renders 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  opacity: 0.3;
  cursor: default;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <button
    className="c1"
    disabled={true}
    onBlur={[Function]}
    onFocus={[Function]}
    type="button"
  />
</div>
`;

exports[`Button focus renders 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  border-color: #FD6FFF;
  box-shadow: 0 0 2px 2px #FD6FFF;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c1 > circle,
.c1 > ellipse,
.c1 > line,
.c1 > path,
.c1 > polygon,
.c1 > polyline,
.c1 > rect {
  outline: #FD6FFF solid 2px;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    <div
      className="c2"
    >
      Test
    </div>
  </button>
</div>
`;

exports[`Button hoverIndicator as object renders 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c1:hover {
  color: #000000;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    hoverIndicator
  </button>
</div>
`;

exports[`Button hoverIndicator as object with color renders 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c1:hover {
  color: #000000;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    hoverIndicator
  </button>
</div>
`;

exports[`Button hoverIndicator as object with colorIndex renders 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c1:hover {
  color: #000000;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    hoverIndicator
  </button>
</div>
`;

exports[`Button hoverIndicator as object with invalid color renders 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c1:hover {
  color: #000000;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    hoverIndicator
  </button>
</div>
`;

exports[`Button hoverIndicator as object with invalid color renders 2`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c1:hover {
  color: #000000;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    hoverIndicator
  </button>
</div>
`;

exports[`Button hoverIndicator as object with invalid colorIndex renders 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c1:hover {
  color: #000000;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    hoverIndicator
  </button>
</div>
`;

exports[`Button hoverIndicator renders 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c1:hover {
  background-color: rgba(221,221,221,0.4);
  color: #444444;
  color: #000000;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    hoverIndicator
  </button>
</div>
`;

exports[`Button href renders 1`] = `
.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <a
    className="c1"
    href="test"
    onBlur={[Function]}
    onFocus={[Function]}
  />
</div>
`;

exports[`Button icon label renders 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 12px;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    <div
      className="c2"
    >
      <svg />
      <div
        className="c3"
      />
      Test
    </div>
  </button>
</div>
`;

exports[`Button primary renders 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background: #7D4CDB;
  color: #f8f8f8;
  border-radius: 18px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    <div
      className="c2"
    >
      Test
    </div>
  </button>
</div>
`;

exports[`Button renders 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    <div
      className="c2"
    >
      Test
    </div>
  </button>
</div>
`;

exports[`Button reverse icon label renders 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 12px;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c1:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    <div
      className="c2"
    >
      Test
      <div
        className="c3"
      />
      <svg />
    </div>
  </button>
</div>
`;

exports[`Button warns about invalid icon render 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
  padding: 12px;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    <div
      className="c2"
    >
      <svg />
    </div>
  </button>
</div>
`;

exports[`Button warns about invalid label render 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    <div
      className="c2"
    >
      Test
    </div>
  </button>
</div>
`;
